hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionContextImpl.java

```diff
@@ -70,7 +70,7 @@ public void beginTransaction() {
 
     @Override
     public void commitTransaction() throws TransactionException {
-        if (transaction.getTransactionType().equals(TWO_PHASE)) {
+        if (transaction.getTransactionType().equals(TWO_PHASE) || transaction.getTransactionLog().size() > 1) {
```

Comment:
Am I stupid or should this be an `&&` instead of `||`? Prepare the commit if it's `TWO_PHASE` AND has more than one entry.